sale_order_field_advanced/models/sale_order_line.py

```diff
@@ -0,0 +1,72 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, fields, api, _
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    adv_reference = fields.Char()
+    adv_is_tailored = fields.Boolean()
+    adv_characteristics = fields.Text()
+    adv_modification_id = fields.Many2one('sale.order.line.modification')
+    adv_note = fields.Text()
+    adv_complement_id = fields.Many2one('product.product')
+    adv_manufacturing_state_id = fields.Many2one('sale.order.line.process.status',
+                                                 compute='_compute_adv_manufacturing_state_id', store=True)
+    adv_date_order = fields.Datetime(
+        related="order_id.date_order",
+        store=True,
+    )
+    adv_product_has_process_control = fields.Boolean(
+        related="product_id.categ_id.has_process_control",
+    )
+
+    adv_manufacturing_state_name = fields.Char(
+        string='Manufacturing State Name',
+        related='adv_manufacturing_state_id.name',
+    )
+
+    adv_manufacturing_state_decoration = fields.Char(compute='_compute_adv_manufacturing_state_decoration')
+
+    def _compute_adv_manufacturing_state_decoration(self):
+        for line in self:
+            if line.adv_manufacturing_state_id:
+                line.adv_manufacturing_state_decoration = line.adv_manufacturing_state_id.decoration_color
+            else:
+                line.adv_manufacturing_state_decoration = False
+
+    @api.depends('product_id')
+    def _compute_adv_manufacturing_state_id(self):
+        for record in self:
+            if record.adv_product_has_process_control:
+                record.adv_manufacturing_state_id = self.env['sale.order.line.process.status'].search(
+                    [("initial_default_state", "=", True)], limit=1)
+            else:
+                record.adv_manufacturing_state_id = self.env['sale.order.line.process.status'].search(
+                    [("no_process_default_state", "=", True)], limit=1)
+                record.clear_fields()
+
+    def clear_fields(self):
+        self.adv_reference = False
+        self.adv_note = False
+        self.adv_modification_id = False
+        self.adv_characteristics = False
+        self.adv_is_tailored = False
+        self.adv_complement_id = False
+
+    def open_manufacturing_state_wizard(self):
+        wizard = self.env['sale.order.line.manufacturing.state.wizard']
+        new = wizard.create({
+            "line_ids": self.env['sale.order.line'].browse(self.env.context.get("active_ids"))})
```

Comment:
This could be simpler, `active_ids` actually have desired ids and `browse` (that requires database access) could be skipped. Not tested, @ChristianSantamaria could you check along with @lperez-solvos ?:
```suggestion
            "line_ids": [(4, line_id) for line_id in self.env.context.get("active_ids", [])],
```